src/posix/platform/rcp_caps_diag.cpp

```diff
@@ -498,22 +502,61 @@ void RcpCapsDiag::SetDiagOutputCallback(otPlatDiagOutputCallback aCallback, void
     mOutputContext  = aContext;
 }
 
+void RcpCapsDiag::ProcessPerf(void) { TestNumSupportedSleepyChildren(); }
+
+void RcpCapsDiag::TestNumSupportedSleepyChildren(void)
+{
+    static constexpr uint16_t kMaxNumChildren = 512;
+    static constexpr uint16_t kValueLength    = 6;
+    otExtAddress              extAddress      = {0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, 0x88};
+    char                      value[kValueLength];
+    uint16_t                  num = 0;
+
+    SuccessOrExit(mRadioSpinel.Set(SPINEL_PROP_MAC_SRC_MATCH_EXTENDED_ADDRESSES, nullptr));
+
+    for (num = 0; num < kMaxNumChildren; num++)
+    {
+        *reinterpret_cast<uint16_t *>(extAddress.m8) = num;
+        SuccessOrExit(
+            mRadioSpinel.Insert(SPINEL_PROP_MAC_SRC_MATCH_EXTENDED_ADDRESSES, SPINEL_DATATYPE_EUI64_S, extAddress.m8));
+    }
+
+exit:
+    if (num != 0)
+    {
+        IgnoreReturnValue(mRadioSpinel.Set(SPINEL_PROP_MAC_SRC_MATCH_EXTENDED_ADDRESSES, nullptr));
+    }
+
+    snprintf(value, sizeof(value), "%u", num);
+    OutputFormat("MaxNumSleepyChildren", value);
```

Comment:
Typically, the source match table capacity for "extended addresses" can be smaller than that for "short addresses."

The "extended address" source match table entry is primarily utilized during the initial attachment process of a sleepy device. Subsequently, on the parent side, we should transition to using "short address" source match table entries for any attached sleepy child.

Reply:
I have changed the code to check both the short address source match table size and the extended address source match table size, and removed the test name `MaxNumSleepyChildren`.